optaplanner-core/src/main/java/org/optaplanner/core/api/score/AbstractBendableScore.java

```diff
@@ -28,13 +28,13 @@
  *
  * @see BendableScore
  */
-public abstract class AbstractBendableScore<S extends Score<S>> extends AbstractScore<S> {
+public abstract class AbstractBendableScore<Score_ extends AbstractBendableScore<Score_>> extends AbstractScore<Score_> {
 
     protected static final String HARD_LABEL = "hard";
     protected static final String SOFT_LABEL = "soft";
     protected static final String[] LEVEL_SUFFIXES = new String[] { HARD_LABEL, SOFT_LABEL };
 
-    protected static String[][] parseBendableScoreTokens(Class<? extends Score> scoreClass, String scoreString) {
+    protected static String[][] parseBendableScoreTokens(Class<? extends Score<?>> scoreClass, String scoreString) {
```

Comment:
I'm not exactly sure of what this method does, but shouldn't the first argument rather be `Class<? extends AbstractBendableScore<?>>`?